[KVConnector] Add metrics to Prometheus-Grafana dashboard

NickLucche opened the pull request:
Follow-up to the work and discussion in this PR https://github.com/vllm-project/vllm/pull/22188.
Here we expose the metrics we're currently tracking for the nixl connector to prometheus.

<img width="1396" height="927" alt="image" src="https://github.com/user-attachments/assets/502e0a19-a66b-49d5-b2ae-0e5caf0bf759" />


cc @markmc  

Update:
@markmc generalized the PR to work for KVConnectorStats more broadly

NickLucche committed:
init

Signed-off-by: NickLucche <nlucches@redhat.com>

markmc committed:
[NIXL][Metrics] Add abstraction for per-connector Prometheus metrics

Signed-off-by: Mark McLoughlin <markmc@redhat.com>

markmc reviewed (changes requested):
Hey @NickLucche 

I wanted to make sure the per-connector metrics was going to work out. I didn't feel good about adding a bunch of NIXL-specific metrics into `vllm/v1/metrics`. So I worked it out at NickLucche/vllm/pull/4

Couple of small inline comments on buckets too :+1:

NickLucche committed:
Merge pull request #4 from markmc/nixl-prometheus-abstraction

[NIXL][Metrics] Add abstraction for per-connector Prometheus metrics

markmc committed:
[KV Connector][Metrics] Add prometheus metrics support to multi-connector

Signed-off-by: Mark McLoughlin <markmc@redhat.com>

markmc committed:
[NIXL][Metrics] Fix NIXL buckets

It's post times that need the smaller bucket size, not
transfer duration.

Uniform 2kb to 16gb range:

```
>>> def human_size(bytes, units=[' bytes','KB','MB','GB','TB', 'PB', 'EB']):
...     """ Returns a human readable string representation of bytes """
...     return str(bytes) + units[0] if bytes < 1024 else human_size(bytes>>10, units[1:])
...
>>> [human_size(2**(10+i)) for i in range(1, 25, 2)]
['2KB', '8KB', '32KB', '128KB', '512KB', '2MB', '8MB', '32MB', '128MB', '512MB', '2GB', '8GB']
```

Signed-off-by: Mark McLoughlin <markmc@redhat.com>

NickLucche committed:
Merge pull request #6 from markmc/nixl-prometheus-buckets

[NIXL][Metrics] Fix NIXL buckets

NickLucche committed:
Merge pull request #5 from markmc/multi-connector-prometheus

[KV Connector][Metrics] Add prometheus metrics support to multi-connector

simon-mo reviewed (approved):
@markmc reviewed

NickLucche committed:
Merge branch 'main' into nixl-prometheus
